draft-ietf-ppm-dap.md

```diff
@@ -484,6 +480,132 @@ struct {
 } HpkeCiphertext;
 ~~~
 
+## Queries {#query}
+
+DAP is designed to give the Collector control over which measurements are
+included in a batch for aggregation. However, since reports are uploaded to the
+Leader, the Collector cannot choose the batch directly. Instead, it issues a "query" used
+by the Aggregators to guide batch selection.
+
+This document defines the following query types:
+
+~~~
+enum {
+   reserved(0), // Reserved for testing purposes
+   time-interval(1),
+   fixed-size(2),
+   (65535)
+} QueryType;
+~~~
+
+Additional query types can be defined as needed. A query includes parameters
+used by the Aggregators to select a given batch of measurements specific to the
+given query type:
+
+~~~
+enum {
+    next-batch(0),
+    by-batch-id(1),
+} FixedSizeQueryType;
```

Comment:
Why not split these out into two separate query types, rather than a single query type with different parameters?

Reply:
I agree that, in the current formulation, the subtypes make sense. However, I don't really see the need for the by-batch-id use case. Why would a collector want to repeatedly query the same batch? Poplar?

Reply:
(Saw the response below -- we can resolve this)